core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -240,6 +240,19 @@ private[spark] object Utils extends Logging {
     // scalastyle:on classforname
   }
 
+  /**
+   * Run a segment of code using a different context class loader in the current thread
+   */
+  def withContextClassLoader[T](ctxClassLoader: ClassLoader)(fn: => T): T = {
+    val oldClassLoader = getContextOrSparkClassLoader
```

Comment:
Curious, why `getContextOrSparkClassLoader` and not just `Thread.getContextClassLoader` ?

Reply:
Originally thought it might be a better idea since the spark executor uses that method when it runs but now that you mention it it could potentially change the behavior...I'll change it back